src/libcollections/vec.rs

```diff
@@ -545,10 +487,10 @@ impl<T> Vec<T> {
                 let p = self.as_mut_ptr().offset(index as isize);
                 // Shift everything over to make space. (Duplicating the
                 // `index`th element into two consecutive places.)
-                ptr::copy(&*p, p.offset(1), len - index);
+                ptr::copy(p, p.offset(1), len - index);
```

Comment:
I read this as some kind of concious thing to make sure it passes a const ptr in the source argument, so that they are not mixed up. Don't think we're going to change ptr::copy again though :)